fof/Platform/Joomla/Platform.php

```diff
@@ -894,6 +894,54 @@ public function logDebug($message)
 		\JLog::add($message, \JLog::DEBUG, 'fof');
 	}
 
+	public function logUserAction($title, $logText, $extension)
```

Comment:
Proposal: add a ```$user``` parameter defaulted to null (the current user) to allow specifying which user did the logging, even in CLI

Reply:
We explicitly prevent CLI from doing any logging because in this case it is not a user taking action. The intent of the User Action Log is to provide an audit log of the actions of logged in users using the web interface.

Reply:
I know. I was suggesting to allow the 3rd party developer to add a user object that he knows made the change. Ie: you're using akeeba backup cli and set that the user that made the change is either the superadmin or a custom user created to log the backup from cli actions

Reply:
I disagree. CLI scripts do not perform positive user authentication. You can *pretend* to be anyone but there is no way to positively know.

In the case of a CLI backup running through CRON I would argue that no human took that backup. Therefore we *must not* attribute this action to a user account (human). Likewise, if the CLI backup script runs it doesn't mean that it was any of the Super Users who did it. It might just as well be your host's support tech who's definitely not a user on your site. Wrongly attributing actions to humans leads to mistrust which defeats the purpose of an audit log.

Reply:
Ok, fair enough.

But what about background jobs? You could argue that the action could be logged when the background job is triggered, but maybe you want to log something else along the execution of the job itself?

Anyway, it was just a suggestion ;)